### PR TITLE
kvm:  ITCO watchdog added

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR, ...

Fixes: #8280 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [x] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
come to think of it, this is probably not the only thing needed. If anyone knows... ?

**Comment by weizhouapache** [comment]
> come to think of it, this is probably not the only thing needed. If anyone knows... ?

@DaanHoogland 
create a RHEL 9 env and test it ?

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alm9 keepEnv

**Review by weizhouapache — APPROVED** [review]
code lgtm

need testing

**Comment by weizhouapache** [comment]
Assigned to myself.  I will test it @DaanHoogland @shwstppr

**Comment by weizhouapache** [comment]
tried to configure watchdog on kvm host, but no success.

anyway, I think this is good to go. According to libvirt, itco is a valid model for watchdog.
https://libvirt.org/formatdomain.html#watchdog-devices

@DaanHoogland 
maybe we could add new action `inject-nmi` as well.

cc @shwstppr

**Comment by DaanHoogland** [comment]
> @DaanHoogland maybe we could add new action `inject-nmi` as well.

seems like a new PR, would you suggest a API/UI enhancement for that? (creating #8368)

**Comment by weizhouapache** [comment]
> > @DaanHoogland maybe we could add new action `inject-nmi` as well.
> 
> seems like a new PR, would you suggest a API/UI enhancement for that? (creating #8368)

no, just a new option for action (which is set by vm.watchdog.action in agent.properties).

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java [hunk 1]
```diff
@@ -2157,7 +2157,7 @@ public String toString() {
 
     public static class WatchDogDef {
         enum WatchDogModel {
-            I6300ESB("i6300esb"), IB700("ib700"), DIAG288("diag288");
+            I6300ESB("i6300esb"), IB700("ib700"), DIAG288("diag288"), ITCO("¨itco");
```
Comment (weizhouapache) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:2160]:
```suggestion
            I6300ESB("i6300esb"), IB700("ib700"), DIAG288("diag288"), ITCO("itco");
```

Comment (weizhouapache) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:2160]: cc @DaanHoogland

Comment (DaanHoogland) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:2160]: i hate me

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
I was not able to test it. (`ITCO` is unsupported on my kvm hosts)

I do not see any error or risk to merge this PR.
what do you think ? @DaanHoogland @rohityadavcloud @shwstppr @JoaoJandre

**Comment by DaanHoogland** [comment]
just to be sure
@blueorangutan test

**Comment by rahultolearn59** [comment]
hi @DaanHoogland, I have experienced the same error with RHEL9 KVM as well with ACS 4.19. Can we get this merged to 4.19 as well?

**Comment by weizhouapache** [comment]
@rahultolearn59 

All commits in 4.18 will be merged into 4.19 and main branch

**Pull request merged**